src/profile-logic/sanitize.js

```diff
@@ -102,9 +102,11 @@ export function sanitizePII(
             return acc;
           }
 
-          counter.mainThreadIndex = newThreadIndex;
           // Adding the current counter to the `counters` list.
-          return acc.concat(counter);
+          return acc.concat({
+            ...counter,
+            mainThreadIndex: newThreadIndex,
+          });
```

Comment:
Ah, I forgot that `concat` can take non-arrays as parameter :)

Maybe we don't need to create a new `acc` here though, we could use `push` and return the same `acc`. It seems overkill to create a new `acc` object for each new element. We can mutate `acc` because we know it's a new array (with `[]` as the second parameter).

If you want that flow warns about mutations, you can explicitely type `counter`: `$ReadOnly<Counter>`.
Maybe we could even do that in our type for Profile: `counters?: $ReadOnlyArray<$ReadOnly<Counter>>`

Reply:
I don't think it's going to be a GC madness to be honest. Since it's an array of objects, all the arrays will share the pointers of those objects and we are going to GC the arrays that contains pointers only(Since we don't remove the elements from the `acc`, we are not GC'ing the objects there). But I'm okay with changing it. We also use the same logic in `threads` field. I will change that too then to make it consistent.

Reply:
Yeah, it depends on how many Counter objects we have. If it's 2 or 3 or even 10 we don't really care indeed :-) I really don't know myself and I assumed we could have many, but maybe it's wrong ?

Reply:
Well it's not too much usually, but it's not a wrong thing to be extra careful :) Updated the code, would you like to take a look?

Reply:
yep, lgtm !